arbnode/transaction_streamer.go

```diff
@@ -244,32 +246,38 @@ func (s *TransactionStreamer) reorg(batch ethdb.Batch, count arbutil.MessageInde
 				// This is the wrong position for the delayed message
 				continue
 			}
-			lastDelayedSeqNum++
-
 			if s.inboxReader != nil {
-				// Verify that the delayed message we're re-sequencing matches what we have after the reorg
-				haveDelayedMessage, err := s.inboxReader.tracker.GetDelayedMessage(delayedSeqNum)
+				// this is a delayed message. Should be resequenced if all 3 agree:
+				// oldMessage, accumulator stored in tracker, and the message re-read from l1
+				expectedAcc, err := s.inboxReader.tracker.GetDelayedAcc(delayedSeqNum)
 				if err != nil {
 					if !strings.Contains(err.Error(), "not found") {
-						log.Error("failed to lookup delayed message to re-sequence", "id", header.RequestId, "err", err)
+						log.Error("reorg-resequence: failed to read expected accumulator", "err", err)
 					}
 					continue
 				}
-				haveDelayedMessageBytes, err := haveDelayedMessage.Serialize()
+				msgBlockNum := new(big.Int).SetUint64(oldMessage.Message.Header.BlockNumber)
+				delayedInBlock, err := s.delayedBridge.LookupMessagesInRange(s.GetContext(), msgBlockNum, msgBlockNum, nil)
 				if err != nil {
-					log.Error("failed to serialize new delayed message from database", "err", err)
+					log.Error("reorg-resequence: failed to serialize old delayed message from database", "err", err)
 					continue
 				}
-				oldDelayedMessageBytes, err := oldMessage.Message.Serialize()
-				if err != nil {
-					log.Error("failed to serialize old delayed message from database", "err", err)
-					continue
+				messageFound := false
+			delayedInBlockLoop:
+				for _, delayedFound := range delayedInBlock {
+					if delayedFound.Message.Header.RequestId.Big().Uint64() != delayedSeqNum {
+						continue delayedInBlockLoop
+					}
+					if expectedAcc == delayedFound.AfterInboxAcc() && delayedFound.Message.Equals(oldMessage.Message) {
+						messageFound = true
+					}
+					break delayedInBlockLoop
```

Comment:
I don't think this loop needs named

Reply:
I agree it's not necessary, but because its inside a different loop with break/continue statements of it's own - naming the internal one improves readability and avoids confusion IMHO.